app/src/lib/stores/app-store.ts

```diff
@@ -444,34 +448,32 @@ export class AppStore extends TypedBaseStore<IAppState> {
   }
 
   private recordTutorialStepCompleted(step: TutorialStep): void {
-    switch (step) {
-      case TutorialStep.PickEditor:
-        this.statsStore.recordTutorialRepoCreated()
-        break
-      case TutorialStep.CreateBranch:
-        this.statsStore.recordTutorialEditorInstalled()
-        break
-      case TutorialStep.EditFile:
-        this.statsStore.recordTutorialBranchCreated()
-        break
-      case TutorialStep.MakeCommit:
-        this.statsStore.recordTutorialFileEdited()
-        break
-      case TutorialStep.PushBranch:
-        this.statsStore.recordTutorialCommitCreated()
-        break
-      case TutorialStep.OpenPullRequest:
-        this.statsStore.recordTutorialBranchPushed()
-        break
-      case TutorialStep.AllDone:
-        this.statsStore.recordTutorialPrCreated()
-        this.statsStore.recordTutorialCompleted()
-        break
-      case TutorialStep.NotApplicable:
-      case TutorialStep.Paused:
-        break
-      default:
-        assertNever(step, 'Unaccounted for step type')
+    // if it's not a valid step, we don't need to record anything
+    if (!isValidTutorialStep(step)) {
+      return
+    }
+
+    const stepInd = orderedTutorialSteps.indexOf(step)
+    this.statsStore.recordHighestTutorialStepCompleted(stepInd)
+
+    if (stepInd > orderedTutorialSteps.indexOf(TutorialStep.PickEditor)) {
+      this.statsStore.recordTutorialEditorInstalled()
```

Comment:
Non-blocking question: So it looks like we'll record this multiple times, right? For example, when the user progresses to step 2 we'll record it, as well as when they progress to steps 3, 4, 5, and 6. Which should be fine, just double-checking that it's intentional/desired. My sense is that we're doing it because it's the only way to account for multiple steps between runs of updateCurrentTutorialStep.